deploy/cloud/modules/ec/variables.tf

```diff
@@ -0,0 +1,76 @@
+variable "ec_api_key" {
+  type = string
+}
+
+variable "stack_version" {
+  description = "Optional version of the Elastic Cloud deployment"
+  type        = string
+  default     = "latest"
+}
+
+variable "region" {
+  description = "Optional region of the Elastic Cloud deployment"
+  type        = string
+  default     = "gcp-us-west2"
+}
+
+variable "deployment_template" {
+  description = "Optional defaults to the CPU optimized template for GCP"
+  type        = string
+  default     = "gcp-compute-optimized-v3"
+}
+
+variable "deployment_name_prefix" {
+  description = "Prefix for the Elastic Cloud deployment name"
+  type        = string
+  default     = "cloud-security"
+}
+
+variable "tags" {
+  type = map(string)
+  default = {
+    "deployment"  = "cloud-security",
+    "environment" = "test-enviroment",
+  }
+  description = "Optional set of tags to use for all deployments"
+}
+
+variable "elasticsearch_size" {
+  default     = "8g"
+  type        = string
+  description = "Optional Elasticsearch instance size"
+}
+
+variable "elasticsearch_zone_count" {
+  default     = 2
+  type        = number
+  description = "Optional Elasticsearch zone count"
+}
+
+variable "elasticsearch_autoscale" {
+  default     = false
+  type        = bool
+  description = "Optional autoscale the Elasticsearch cluster"
+}
+
+# Docker image overrides
+
+variable "docker_image_tag_override" {
+  default = {
+    "elasticsearch" = "",
+    "kibana"        = "",
+    "apm"           = "",
```

Comment:
Do we need those?

Reply:
please add comment about the override logic and why do we must state an override for BC/